### PR TITLE
chore: prepare for `v0.1.2` release

### PR DESCRIPTION
**Pull request by vlopes11**
```[tasklist]

### Must have

- [x] Update `CHANGELOG.md`
- [x] Increment version
```

```[tasklist]

### Nice to have

- [x] Improve `README.md` documentation, listing the new types
```

```[tasklist]

### Tech-debts introduced
- [ ] #65  
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -12,8 +12,15 @@ For performance benchmarks of these hash functions and their comparison to other
 ## Merkle
 [Merkle module](./src/merkle/) provides a set of data structures related to Merkle trees. All these data structures are implemented using the RPO hash function described above. The data structures are:
 
-* `MerkleTree`: a regular fully-balanced binary Merkle tree. The depth of this tree can be at most 64.
+* `MerkleError`: a wrapper for the possible error variants when performing Merkle tree operations, including storage failure.
+* `MerklePath`: a Merkle path container, composed of a sequence of nodes. This wrapper struct will encapsulate the Merkle opening verification.
 * `MerklePathSet`: a collection of Merkle authentication paths all resolving to the same root. The length of the paths can be at most 64.
+* `MerkleTree`: a regular fully-balanced binary Merkle tree. The depth of this tree can be at most 64.
+* `NodeIndex`: a Merkle tree address for it's nodes, consisting of a tuple depth+value.
```
Comment (grjte) [inline README.md:19]: nit: this should be "... for its nodes..." Also, I would probably say "...consisting of a tuple containing `depth` and `value`."

File: src/merkle/simple_smt/mod.rs [hunk 2]
```diff
@@ -6,7 +6,7 @@ mod tests;
 // SPARSE MERKLE TREE
 // ================================================================================================
 
-/// A sparse Merkle tree with 63-bit keys and 4-element leaf values, without compaction.
+/// A sparse Merkle tree with 64-bit keys and 4-element leaf values, without compaction.
```
Comment (grjte) [inline src/merkle/simple_smt/mod.rs:9]: Sorry I missed this - when did we switch from 63-bit to 64-bit keys. Do you happen to have a reference to the discussion?

**Review by grjte — CHANGES_REQUESTED** [review]
Thanks @vlopes11! I left one minor comment inline and a question

File: README.md (hunk 1 above)
Comment (vlopes11) [reply README.md]: Fixed!

File: src/merkle/simple_smt/mod.rs (hunk 2 above)
Comment (vlopes11) [reply src/merkle/simple_smt/mod.rs]: There isn't a discussion over that. The idea is to make it consistent with the RPO module documentation and arguments of the `SimpleSmt` (that will take a `u64`).

https://github.com/0xPolygonMiden/crypto/blob/398af59045efe16181140475742c7535618a9c75/src/hash/rpo/mod.rs#L60-L61

I'm reverting to the previous `63` so we can retake this topic for the next release.

Comment (vlopes11) [reply src/merkle/simple_smt/mod.rs]: Added #65

**Review by grjte — APPROVED** [review]
LGTM, thanks @vlopes11!

File: README.md [hunk 3]
```diff
@@ -12,8 +12,15 @@ For performance benchmarks of these hash functions and their comparison to other
 ## Merkle
 [Merkle module](./src/merkle/) provides a set of data structures related to Merkle trees. All these data structures are implemented using the RPO hash function described above. The data structures are:
 
-* `MerkleTree`: a regular fully-balanced binary Merkle tree. The depth of this tree can be at most 64.
+* `MerkleError`: a wrapper for the possible error variants when performing Merkle tree operations, including storage failure.
+* `MerklePath`: a Merkle path container, composed of a sequence of nodes. This wrapper struct will encapsulate the Merkle opening verification.
 * `MerklePathSet`: a collection of Merkle authentication paths all resolving to the same root. The length of the paths can be at most 64.
+* `MerkleTree`: a regular fully-balanced binary Merkle tree. The depth of this tree can be at most 64.
+* `NodeIndex`: a Merkle tree address for its nodes, consisting of a tuple containing `depth` and `value`.
+* `SimpleSmt`: a Sparse Merkle Tree, mapping 63-bit keys to 4-element leaf values.
```
Comment (bobbinth) [inline README.md:20]: I would probably either remove `MerkleError`, `NodeIndex`, and `MerklePath` or maybe just mention them in a single sentence under this list. These are supporting structs which are used by `MerkleTree`, `SimpleSmt`, and `MerklePathSet`, and I don't think we need to call as much attention to them.

Comment (vlopes11) [reply README.md]: Done!

Comment (bobbinth) [reply README.md]: Thank you! Sorry for another request, but could we change this to something like this:

* `MerkleTree`: a regular fully-balanced binary Merkle tree. The depth of this tree can be at most 64.
* `SimpleSmt`: a Sparse Merkle Tree, mapping 63-bit keys to 4-element leaf values.
* `MerklePathSet`: a collection of Merkle authentication paths all resolving to the same root. The length of the paths can be at most 64.

The module also contains additional supporting components such as `NodeIndex`, `MerklePath`,  and `MerkleError`  to assist with tree indexation, opening proofs, and reporting inconsistent arguments/state.

**Review by bobbinth — APPROVED** [review]

**Pull request merged**